In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import warnings
from math import factorial

## Урок 4. Непрерывные случайные величины. Функция распределения и плотность распределения вероятностей. Равномерное и нормальное распределение. Центральная предельная теорема

### 1.
Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800]. Найдите ее среднее значение и дисперсию.

In [3]:
M = (200+800)/2
print('Среднее значение', M)

Среднее значение 500.0


In [4]:
D = ((800-200)**2)/12
print('Дисперсия ', D)

Дисперсия  30000.0


### 2.
О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

D = ((b-a)**2)/12
12D = b**2 -2*b*a + a**2
2.4 = b**2-2b*a +0.25
2.15=b**2-2*b*a
b**2-2*b*a - 2.15 =0
b**2-2*b*0.5 - 2.15 =0
b**2-b - 2.15 =0
Diskr = 1**2 +4*1*2.15
Diskr = 9.6
x1,2 = (-b +- sqrt(Diskr) )/2a
x1 = (1 + (1+4*1*2.15)**0.5)/2
x2 = (1 - (1+4*1*2.15)**0.5)/2


In [5]:
x1 = (1 + (1+4*1*2.15)**0.5)/2
x2 = (1 - (1+4*1*2.15)**0.5)/2
print(x1,x2)

2.049193338482967 -1.0491933384829668


-1.0491933384829668 не подходит, поскольку мы знаем что левая гравница это 0.5


In [6]:
M = (2.049193338482967 + 0.5)/2
print('Ответ, правая граница величины B -', x1, 'Среднее значение - ', M)

Ответ, правая граница величины B - 2.049193338482967 Среднее значение -  1.2745966692414834


### 3.
Непрерывная случайная величина X распределена нормально и задана плотностью распределения
f(x) = (1 / (4 * sqrt(2*pi))) * (exp(-((x+2)**2) / 32))
Найдите:
а). M(X)
б). D(X)
в). std(X) (среднее квадратичное отклонение)

так как мы знаем что Х распределена нормально -  M(x) = -2, sigma = 4, D(X) = 16

##№ 4.
Рост взрослого населения города X имеет нормальное распределение.
Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.
Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
а). больше 182 см
б). больше 190 см
в). от 166 см до 190 см
г). от 166 см до 182 см
д). от 158 см до 190 см
е). не выше 150 см или не ниже 190 см
ё). не выше 150 см или не ниже 198 см
ж). ниже 166 см.


а) P(X>182)= Ф(+∞) - Ф(182)

Ф - функция Лапласа. Свойство функции:

1. Ф(0) = 0
2. Ф(−x) = -Ф(x)
3. Ф(+∞) = 1/2
4. Ф(−∞) = − 1/2

Ф(+∞) - Ф(182) = 1/2 - (Ф((182-174)/8) = 1/2 - Ф(1) = 1/2 - 0,3413

In [18]:
1/2 - 0.3413

0.1587

б)P(X>190)=Ф(+∞) - Ф(190)

Ф(+∞) - Ф(190) = 1/2 - (Ф((190-174)/8) = 1/2 - Ф(2) = 1/2 - 0,4772

In [29]:
1/2 - 0.4772

0.022799999999999987

в)P(166<X<190)= Ф(190) - Ф(166) = Ф(190-174/8) - Ф(166-174/8) = Ф(2)+Ф(1)

In [9]:
0.4772 + 0.3413

0.8185

г) P(166<X<182)= Ф(1)+Ф(1)

In [10]:
0.3413+0.3413

0.6826

д) P(158<X<190)= Ф(190) - Ф(158) = Ф(2)+Ф(2)

In [11]:
0.4772+0.4772

0.9544

e)  P(X<150)=F(150) = Ф(150) - Ф(-∞) = -Ф(3) + 1/2

In [37]:
0.5-0.49865


0.0013500000000000179

P(X>190)=0.0228

In [38]:
print('не выше 150 см или не ниже 190 см', 0.00135+0.0228)

не выше 150 см или не ниже 190 см 0.02415


ё) P(X<150) = Ф(150-174/8) - Ф(-∞)= -0.49865 + 1/2 = 0.00135; P(X>198) = Ф(+∞) - Ф(198)= 1/2 - Ф(3) = 0.00135

In [19]:
0.00135 + 0.00135

0.0027

ж) P(X<166) = Ф(166-174/8) - Ф(-∞) = -Ф(1)+1/2    (Аналогично заданию а)

In [16]:
-0.3413+1/2

0.1587

## Задание 5 На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

In [17]:
(190-178)/25**0.5

2.4

Ответ на 2.4 сигмы